src/app/dashboard/commodities/[guid]/page.tsx

```diff
@@ -56,6 +57,7 @@ export default function CommodityPage({ params }: CommodityPageProps): JSX.Eleme
       byCurrency[price.currency.guid].push(price);
     });
   }
+  const commodityCurrency = commodity.namespace !== 'CURRENCY' ? prices?.[0].fk_currency || undefined : undefined;
```

Comment:
For non CURRENCY commodities, we always have the same currency assigned (i.e. STOCKS only have one currency). This takes the first currency for a stock and passes it as a default